sys/arm64/arm64/unwind.c

```diff
@@ -41,12 +41,13 @@ unwind_frame(struct thread *td, struct unwind_state *frame)
 
 	fp = frame->fp;
 
-	if (!kstack_contains(td, fp, sizeof(uintptr_t) * 2))
+	if (!is_aligned(fp, sizeof(fp)) ||
+	    !kstack_contains(td, fp, sizeof(fp) * 2))
 		return (false);
 
 #ifdef __CHERI_PURE_CAPABILITY__
 	if ((ptraddr_t)fp < cheri_getbase(fp) ||
-	    (ptraddr_t)(fp + sizeof(uintptr_t) * 2) >= cheri_gettop(fp) ||
+	    (ptraddr_t)(fp + sizeof(fp) * 2) > cheri_gettop(fp) ||
```

Comment:
You can leave the sizeof(uintptr_t) (it doesn't hurt and is always correct).  I would do the `>=` to '>' change as a separate commit.  Then the first commit that just adds the `is_aligned` check can be upstreamed directly (assuming is_aligned is in upstream already)

Reply:
OK, I'll split the commits.

I changed `sizeof(uintptr_t)` so that it's consistent with the RISC-V implementation, but I'm happy to change it back.